modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationReturnTypeModelValidator.java

```diff
@@ -123,4 +143,13 @@ private IllegalModelDefinitionException missingReturnTypeException(ExtensionMode
     throw new IllegalOperationModelDefinitionException(format("Operation '%s' in Extension '%s' is missing a return type",
                                                               operationModel.getName(), model.getName()));
   }
+
+  private void validateGenerics(ExtensionModel extensionModel, NamedObject namedObject, OperationModel operationModel,
```

Comment:
why is the `namedObject` parameter needed? Seems redundant. This method should operate over the operationModel entirely

Reply:
In the case of the CompletionCallback it also operates over the operation's extensionParameter. I can use the operationModel on the ProblemReporter but I didn't want to change how it was being used(when there is no generics for example).